usaspending_api/search/v2/views/search.py

```diff
@@ -158,10 +158,45 @@ def post(self, request):
         if filters is None:
             raise InvalidParameterException("Missing one or more required request parameters: filters")
 
-        # filter queryset
-        queryset = transaction_filter(filters)
+        # build sql query filters
+        if can_use_view(filters, 'SummaryView'):
+            potential_scopes = ["agency", "subagency"]
+            if scope not in potential_scopes:
+                raise InvalidParameterException("scope does not have a valid value")
+            if scope == 'agency':
+                queryset = view_filter(filters, 'SummaryView')
+            else:
+                queryset = view_filter(filters, 'SummarySubagencyView')
+
+            if category == "awarding_agency":
+                queryset = queryset \
+                    .filter(awarding_agency_name__isnull=False) \
+                    .values(
+                        agency_name=F('awarding_agency_name'),
+                        agency_abbreviation=F('awarding_agency_abbr')) \
+                    .annotate(aggregated_amount=Sum('federal_action_obligation')) \
+                    .order_by('-aggregated_amount')
+            elif category == "funding_agency":
+                queryset = queryset \
+                    .filter(funding_agency_name__isnull=False) \
+                    .values(
+                        agency_name=F('funding_agency_name'),
+                        agency_abbreviation=F('funding_agency_abbr')) \
+                    .annotate(aggregated_amount=Sum('federal_action_obligation')) \
+                    .order_by('-aggregated_amount')
```

Comment:
Very similar code here, would recommend something like:
```
agency_category = None
if category in ["awarding_agency", "funding_agency"]:
    agency_filters = {"{}_name__isnull".format(category): False}
    agency_values = {"agency_name": F('{}_name'.format(category)),
                     "agency_abbreviation": F('{}_abbr'.format(category))}
    queryset = queryset \
        .filter(**agency_filters) \
        .values(**agency_values) \
        .annotate(aggregated_amount=Sum('federal_action_obligation')) \
        .order_by('-aggregated_amount')
```
But then again, that'd effect the whole file as well so it's not crucial. Just something to keep in mind for this file.

Reply:
Perhaps we can look at stuff like this when we can refactor. It adds unnecessary complexity when there are small differences.